tests/table-operations.spec.ts

```diff
@@ -0,0 +1,495 @@
+  });
+
+  await test.step('Toggle Visibility', async () => {
+    await page
+      .getByRole('button', { name: 'Toggle visibility Double' })
+      .getByRole('button', { name: 'Toggle visibility' })
+      .click();
+
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+  });
+});
+
+test('can custom column', async ({ page }) => {
+  await page.locator('data-testid=menu-item-Custom Columns').click();
+  await expect(page.getByText('Table Options')).toHaveCount(0);
+
+  await test.step('Create custom column', async () => {
+    const columnName = page.getByPlaceholder('Column Name');
+    await expect(columnName).toHaveCount(1);
+    await columnName.click();
+    await page.keyboard.type('Test');
+
+    const columnFormula = page.locator('.editor-container');
+    await expect(columnFormula).toHaveCount(1);
+    await columnFormula.click();
+    await page.keyboard.type('Double * 2');
+  });
+
+  const addColumnButton = page.getByRole('button', {
+    name: 'Add Another Column',
+  });
+  const saveButton = page.getByRole('button', { name: 'Save Column' });
+
+  await test.step('Create 2nd custom columns from 1st', async () => {
+    await addColumnButton.click();
+
+    const newColumnName = page.getByPlaceholder('Column Name').nth(1);
+    await newColumnName.click();
+    await page.keyboard.type('Test2');
+
+    const newColumnFormula = page.locator('.editor-container').nth(1);
+    await newColumnFormula.click();
+    await page.keyboard.type('Test * 2');
+
+    await saveButton.click();
+
+    await waitForLoadingDone(page);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+  });
+
+  await test.step('Delete', async () => {
+    const deleteLastColumnButton = page
+      .getByRole('button', { name: 'Delete custom column' })
+      .nth(1);
+    await deleteLastColumnButton.click();
+    await saveButton.click();
+
+    await waitForLoadingDone(page);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+  });
+
+  await test.step('Drag', async () => {
+    await addColumnButton.click();
+
+    const dragColumn = page.getByPlaceholder('Column Name').nth(1);
+    await dragColumn.click();
+    await page.keyboard.type('Drag');
+
+    const dragColumnFormula = page.locator('.editor-container').nth(1);
+    await dragColumnFormula.click();
+    await page.keyboard.type('String');
+
+    const dragButton = page
+      .getByRole('button', { name: 'Drag column to re-order' })
+      .nth(1);
+    const panelAbove = page
+      .getByRole('button', { name: 'Drag column to re-order' })
+      .first();
+    const dropIndicator = page
+      .locator('.custom-column-builder-container')
+      .locator('.dragging');
+
+    const browser = dragButton.page().context().browser()?.browserType().name();
+    await dragComponent(
+      dragButton,
+      panelAbove,
+      dropIndicator,
+      0,
+      browser === 'webkit' ? 1000 : undefined
```

Comment:
Still couldn't figure out the work around for this. The coordinates that it was being dragged to looked correct so I don't think it was the same issue that we saw yesterday where it was out of the page bounds because the menu animation didn't finish. Can I just mark this as a TODO and create a ticket for it?

Reply:
Yes mark TODO, add a ticket, and attach a video of a run that's failing where it looks correct.